orange/src/main/java/com/hartwig/hmftools/orange/algo/OrangeAlgo.java

```diff
@@ -252,30 +252,62 @@ public OrangeRecord run(@NotNull OrangeConfig config) throws Exception
-        OrangeRecord report = ImmutableOrangeRecord.builder()
-                .sampleId(config.tumorSampleId())
-                .samplingDate(config.samplingDate())
-                .experimentType(config.experimentType())
-                .configuredPrimaryTumor(ConversionUtil.mapToIterable(configuredPrimaryTumor, OrangeConversion::convert))
-                .refGenomeVersion(config.refGenomeVersion())
-                .pipelineVersion(pipelineVersion)
-                .refSample(refSample)
-                .tumorSample(tumorSample)
-                .germlineMVLHPerGene(mvlhPerGene)
-                .purple(purple)
-                .linx(linx)
-                .wildTypeGenes(wildTypeGenes)
-                .isofox(isofox)
-                .lilac(lilac != null ? OrangeConversion.convert(lilac, hasRefSample, config.rnaConfig() != null) : null)
-                .immuneEscape(immuneEscape)
-                .virusInterpreter(virusInterpreter != null ? VirusInterpreter.interpret(virusInterpreter) : null)
-                .chord(chord != null ? OrangeConversion.convert(chord) : null)
-                .cuppa(cuppa)
-                .peach(ConversionUtil.mapToIterable(peach, OrangeConversion::convert))
-                .sigAllocations(SigsInterpreter.interpret(sigAllocations, etiologyPerSignature))
-                .cohortEvaluations(evaluateCohortPercentiles(config, purple))
-                .plots(buildPlots(config))
-                .build();
+        OrangeRecord report;
+        if (peach != null)
+        {
+            report = ImmutableOrangeRecord.builder()
+                    .sampleId(config.tumorSampleId())
+                    .samplingDate(config.samplingDate())
+                    .experimentType(config.experimentType())
+                    .configuredPrimaryTumor(ConversionUtil.mapToIterable(configuredPrimaryTumor, OrangeConversion::convert))
+                    .refGenomeVersion(config.refGenomeVersion())
+                    .pipelineVersion(pipelineVersion)
+                    .refSample(refSample)
+                    .tumorSample(tumorSample)
+                    .germlineMVLHPerGene(mvlhPerGene)
+                    .purple(purple)
+                    .linx(linx)
+                    .wildTypeGenes(wildTypeGenes)
+                    .isofox(isofox)
+                    .lilac(lilac != null ? OrangeConversion.convert(lilac, hasRefSample, config.rnaConfig() != null) : null)
+                    .immuneEscape(immuneEscape)
+                    .virusInterpreter(virusInterpreter != null ? VirusInterpreter.interpret(virusInterpreter) : null)
+                    .chord(chord != null ? OrangeConversion.convert(chord) : null)
+                    .cuppa(cuppa)
+                    .peach(ConversionUtil.mapToIterable(peach.stream().filter(genotype -> !genotype.gene().equals("UGT1A1")).toList(), OrangeConversion::convert))
```

Comment:
I would suggest creating a method/function that takes the set of peach genotype as input and:
- checks for the null case
- does the filtering.

It's a bit more consistent with the rest of the class, which breaks out a method and variable when the conversion becomes more than one line (eg. loadCuppaData).